aea/connections/stub/connection.py

```diff
@@ -77,7 +76,7 @@ def _decode(e: bytes, separator: bytes = SEPARATOR):
     # protobuf messages cannot be delimited as they can contain an arbitrary byte sequence; however
     # we know everything remaining constitutes the protobuf message.
     message = SEPARATOR.join(split[3:-1])
-    message = codecs.decode(message, "unicode-escape").encode("utf-8")
+    # message = codecs.decode(message, "unicode-escape").encode("utf-8")  # noqa: E800
```

Comment:
For `test_echo`, This line added a new byte `\xc2` at position 1:

```python
b"\x12\x87\x01\x08\x01\x12@aa88dbbc4ab9ad89aff8682b4474a241055279e40f628aa263a8cd3848672eac*A\n4type.googleapis.com/fetch.aea.Default.DefaultMessage\x12\t*\x07\n\x05hello"

b"\x12\xc2\x87\x01\x08\x01\x12@aa88dbbc4ab9ad89aff8682b4474a241055279e40f628aa263a8cd3848672eac*A\n4type.googleapis.com/fetch.aea.Default.DefaultMessage\x12\t*\x07\n\x05hello"
```

The first is the byte encoding of a default message; the second is the message read by the stub connection before the change.

Reply:
Not sure what's going on here. Looks like we finally need to address https://github.com/fetchai/agents-aea/issues/1413

Reply:
We cannot merge before we have addressed #1413..not safe imo

Reply:
The main limiting factor with the stub connection input file parsing is the need to support `echo`ing a serialized message in text format. I believe that providing an `aea` cli helper that constructs a message from command line arguments and serialize it to the stdout will free the stub connection to use any encoding/decoding format.

Reply:
We already have `aea interact` command. What's missing in the stub connection is specifying the length of the bytes string in the encoding so we can do away with the hack.